tests/components/airly/test_config_flow.py

```diff
@@ -0,0 +1,97 @@
+"""Define tests for the Airly config flow."""
+import json
+
+from asynctest import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.components.airly import config_flow
+from homeassistant.components.airly.const import CONF_LANGUAGE, DOMAIN
+
+from tests.common import MockConfigEntry, load_fixture
+
+CONFIG = {
+    CONF_NAME: "abcd",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 123,
+    CONF_LONGITUDE: 456,
+    CONF_LANGUAGE: "en",
+}
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = config_flow.AirlyFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_show_form_with_input(hass):
+    """Test that the form is served with input."""
+    flow = config_flow.AirlyFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=CONFIG)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_invalid_api_key(hass):
+    """Test that errors are shown when API_KEY is invalid."""
+    flow = config_flow.AirlyFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=CONFIG)
```

Comment:
Won't this make a request? We have to patch the library so that no real requests, ie no I/O, is done in any tests.

Reply:
@MartinHjelmare Fixed.